src/Combinatorics/Matroids/realization_space.jl

```diff
@@ -0,0 +1,989 @@
+
+
+mutable struct MatroidRealizationSpace
+    defining_ideal::Union{Ideal, NumFieldOrdIdl, Nothing}
+    inequations::Union{Vector{Oscar.RingElem},Nothing}
+    ambient_ring::Union{Oscar.MPolyRing, Ring, Nothing}
+    realization_matrix::Union{Oscar.MatElem,Nothing}
+    realizable::Union{Bool,Nothing} # Nothing if we're not sure yet if M is realizable.
+    F::AbstractAlgebra.Ring
+    char::Union{Int,Nothing}
+    q::Union{Int,Nothing}
+    one_realization::Bool
+end
+
+function Base.show(io::IO, RS::MatroidRealizationSpace)
+    if RS.realizable == false
+        if RS.char == nothing && RS.q == nothing && RS.F == ZZ
+            print(io, "The matroid is not realizable.")
+        else
+            print(io, "The matroid is not realizable over the specified field or characteristic.")
+        end
+    else
+        if RS.one_realization
+            println(io, "One realization is given by")
+        else
+            println(io, "The realizations are parametrized by")
+        end
+        # println isn't ideal as it prints the matrix as one big line
+        display(RS.realization_matrix)
+        println(io, "in the ", RS.ambient_ring)
+        I = RS.defining_ideal
+        if (typeof(I) <: NumFieldOrdIdl && I.gen != ZZ(0)) || (typeof(I) <: Ideal && !iszero(I))
+            println(io, "within the vanishing set of the ideal\n", RS.defining_ideal)
+        end
+        if length(RS.inequations) > 0
+            println(io, "avoiding the zero loci of the polynomials\n", RS.inequations)
+        end
+    end
+end
+
+# Constructors
+function MatroidRealizationSpace(F::AbstractAlgebra.Ring, char::Union{Int,Nothing}, q::Union{Int,Nothing})
+    return MatroidRealizationSpace(nothing, nothing, nothing, nothing, false, F, char, q, false)
+end
+
+function MatroidRealizationSpace(I::Union{Ideal, NumFieldOrdIdl, Nothing},
```

Comment:
The above two constructors probably should be inner constructors. It is also weird to ask for arguments of type `nothing` like this -- it might make sense if you wish to provide default values for these arguments, but like this is it is strange.

Reply:
Fixed.